subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNodeExecutor.java

```diff
@@ -189,25 +191,34 @@ private static boolean missesDependency(Node producer, Node consumer) {
         // Do a breadth first search for any dependency
         Deque<Node> queue = new ArrayDeque<>();
         Set<Node> seenNodes = new HashSet<>();
-        consumer.getHardSuccessors().forEach(successor -> {
-            if (seenNodes.add(successor)) {
-                queue.add(successor);
-            }
-        });
+        addHardSuccessorTasksToQueue(consumer, queue, seenNodes);
         while (!queue.isEmpty()) {
             Node dependency = queue.removeFirst();
             if (dependency == producer) {
                 return false;
             }
-            dependency.getHardSuccessors().forEach(node -> {
-                if (seenNodes.add(node)) {
-                    queue.add(node);
-                }
-            });
+            addHardSuccessorTasksToQueue(dependency, queue, seenNodes);
         }
         return true;
     }
 
+    private static void addHardSuccessorTasksToQueue(Node node, Queue<Node> queue, Set<Node> seenNodes) {
```

Comment:
I wonder if you need this method at all. Wouldn't calling:

```
findHardSuccessorTasks(consumer, successor -> {
                if (seenNodes.add(taskSuccessor)) {
                    queue.add(taskSuccessor);
                }
});
```
have the same effect?
```

Reply:
It wouldn't, since `consumer` is already a `LocalTaskNode`. So it wouldn't go into the dependencies at all.

Reply:
Though I suppose I can turn around the check, let me try.

Reply:
Better now?

Reply:
Ah yes, I knew there was something weird with having to use 2 methods :) :+1: